pkg/util/pki/common.go

```diff
@@ -203,6 +203,42 @@ func BrokerUserForCluster(cluster *v1beta1.KafkaCluster, extListenerStatuses map
 	}
 }
 
+// BrokerUserForClusterWithPKIBackendSpec returns a KafkaUser CR with PKIBackendSpec included for the broker certificates in a KafkaCluster
+func BrokerUserForClusterWithPKIBackendSpec(cluster *v1beta1.KafkaCluster, extListenerStatuses map[string]v1beta1.ListenerStatusList) *v1alpha1.KafkaUser {
+	additionalHosts := make([]string, 0, len(extListenerStatuses))
+	for _, listenerStatus := range extListenerStatuses {
+		for _, status := range listenerStatus {
+			host := strings.Split(status.Address, ":")[0]
+			additionalHosts = append(additionalHosts, host)
+		}
+	}
+	additionalHosts = sortAndDedupe(additionalHosts)
+	sslConfig := cluster.Spec.ListenersConfig.SSLSecrets
+	var pkiBackend string
+	if sslConfig.PKIBackend != "" {
+		pkiBackend = string(sslConfig.PKIBackend)
+	} else {
+		pkiBackend = string(v1beta1.PKIBackendCertManager)
+	}
+	return &v1alpha1.KafkaUser{
+		ObjectMeta: templates.ObjectMeta(EnsureValidCommonNameLen(GetCommonName(cluster)), LabelsForKafkaPKI(cluster.Name, cluster.Namespace), cluster),
+		Spec: v1alpha1.KafkaUserSpec{
+			SecretName: fmt.Sprintf(BrokerServerCertTemplate, cluster.Name),
+			DNSNames:   append(GetInternalDNSNames(cluster), additionalHosts...),
+			IncludeJKS: true,
+			ClusterRef: v1alpha1.ClusterReference{
+				Name:      cluster.Name,
+				Namespace: cluster.Namespace,
+			},
+			PKIBackendSpec: &v1alpha1.PKIBackendSpec{
+				// Not checking IssuerRef for nil as it is checked in caller function kafkapki
+				IssuerRef:  sslConfig.IssuerRef,
+				PKIBackend: pkiBackend,
```

Comment:
Currently this can be cert-manager only so this modification useless.
We need to extend the k8s api validation here: https://github.com/shubhamcoc/koperator/blob/2c2621944be96aa162024aab581807d4b3f56967/api/v1beta1/kafkacluster_types.go#L491
It need to be "k8s-csr,cert-manager"
Do we really want to this?
If yes then we have to test that case when pkiBackend is k8s-csr.
Currently when the sslConfig.Create is false (https://github.com/banzaicloud/koperator/blob/84c46436b75029b2d4dffa09b80080fb85a0eea3/pkg/pki/certmanagerpki/certmanager_pki.go#L111) then  userProvidedPKI function will be executed.
In this case a new Cert-manager issuer will be created ( https://github.com/banzaicloud/koperator/blob/84c46436b75029b2d4dffa09b80080fb85a0eea3/pkg/pki/certmanagerpki/certmanager_pki.go#L237) which is unnecessary when pkibackend is k8s-csr.

Reply:
I can do the test after the fixes.

Reply:
I did the changes, but since we are changing in the api pkg, we might need to update it in go.mod to use local api.

Reply:
We need to create a separate PR for the API change and we need a new tag for the API when it is merged to the master branch. This new tag should be used in this PR .

Reply:
You can use Go workspaces for local development before putting up the final PRs for the different modules.

Reply:
updated the go.mod file

Reply:
If my understanding is correct (see [here](https://github.com/banzaicloud/koperator/issues/911#issuecomment-1546228732)), believe the original request is to ask for support to customize the `group` field with `cert-manager`, instead of asking support for `k8s-csr` - which is good to have tho but I don't think supporting `k8s-csr` is what this PR is aiming for

Reply:
Yes @panyuenlau you are right, but if the issuerRef is present with create option as true, do we create kafka-server or kafka-client certificates??